lib/tasks/allrecipe-init.rake

```diff
@@ -0,0 +1,172 @@
+task :init, [:start_frequency] => :environment do |task, args|
+  start_time = Time.now
+  require 'open-uri'
+  require 'nokogiri'
+  start_frequency = args.start_frequency.to_i
+  end_frequency = start_frequency + 2
+
+  for i in start_frequency..end_frequency
+    puts "page " + i.to_s
+    @tries = 0
+    begin
+      url = "http://allrecipes.com/recipes/main.aspx?vm=l&Page="+i.to_s
+      browser = open(url).read
+      html_doc = Nokogiri::HTML(browser)
+
+      #gather all information into arrays
+      descriptions = html_doc.css('div > div > div > div > span.popReview')
+      url_code_elements = html_doc.css('div > div > div > h3 > a')
+      authors = html_doc.css('div.searchResult.hub-list-view > div.search_mostPop > div.searchRtSd > div.search_mostPop_Review > span:nth-child(2)')
+
+      #loop each author and check if OutsideProfile exists, if not, create one and create new_recipe
+      authors.each_with_index do |author, index|
+        #check if OutsideProfile Exists, if not, create one and continue
+        if author.css('a').any?
+          author_name = author.css('a').text.squish
+          author_href_array = author.css('a').attr('href').to_s.split('/')
+          outside_profile_url = "http://allrecipes.com/cook/" + author_href_array[4]
+        else
+          author_name = author.text.squish
+          outside_profile_url = "http://allrecipes.com/"
+        end
+        author_existence = OutsideProfile.find_by(username: author_name, outside_profile_url: outside_profile_url)
+        if author_existence == nil
+          author_existence = OutsideProfile.create(display_format: 1, username: author_name, outside_profile_url: outside_profile_url)
+        end
+
+        #extract information and putting it in new_recipe
+        url = url_code_elements[index].attr('href')
+        url_array = url.split('/')
+
+        if Recipe.find_by(url_code: url_array[4], domain_name_id: 1) == nil
+          #create a new recipe entry
+          new_recipe = author_existence.recipes.new
+
+          new_recipe.outside_profile_id = author_existence.id
+          new_recipe.domain_name_id = 1.to_i
+          new_recipe.url_code = url_array[4]
+          new_recipe.name = url_code_elements[index].text
+          new_recipe.description = descriptions[index].text
+
+          #go to part two of scrapping process to scrape recipe in depth information
+          get_recipe_indepth_info(url, new_recipe)
```

Comment:
I don't like how `new_recipe.save` is nested inside another function. It's so hard to follow and find. Maybe you can save general information first, and then you have separate task that will scrape in-depth information. It will be easier for you to manage scraping.